### PR TITLE
[MRG] update documentation & add notebooks to docs

### LINKED ISSUE #430: revisit jupyter demo notebooks / binder stuff before 2.0 release
haven't run 'em in a while

### PR DESCRIPTION
**Pull request by ctb**
This reorganizes the documentation and adds Jupyter notebooks + binder links. See

* [top level index](https://sourmash.readthedocs.io/en/add-nbsphinx/)
* [tutorials](https://sourmash.readthedocs.io/en/add-nbsphinx/tutorials.html)

for the major points of reorg.

Rendered notebooks are under the [add-nbsphinx](https://sourmash.readthedocs.io/en/add-nbsphinx/) branch on readthedocs. Direct links:

* [k-mers and minhash](https://sourmash.readthedocs.io/en/add-nbsphinx/kmers-and-minhash.html)
* [sourmash examples](https://sourmash.readthedocs.io/en/add-nbsphinx/sourmash-examples.html)
* [sourmash on collections](https://sourmash.readthedocs.io/en/add-nbsphinx/sourmash-collections.html)

both tutorials are also now updated to work, and to use bioconda:

* [basic tutorial](https://sourmash.readthedocs.io/en/add-nbsphinx/tutorial-basic.html)
* [LCA tutorial](https://sourmash.readthedocs.io/en/add-nbsphinx/tutorials-lca.html)

Fixes #430 #552. See #628 for related discussions. Includes cleanup from #607.

TODO:
- [x] link these into the rest of the documentation :)
- [x] add binder integration so that from the documentation, you can directly click to execute notebooks
- [x] add relative links between notebooks so that once in binder, you don't have to start up an independent binder
- [ ] before merge, check to make sure that we're pointing binder at the right branch (master, presumably)

Checklist:
- [ ] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [x] `make coverage` Is the new code covered?
- [x] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

**Comment by ctb** [comment]
w00t the basic config seems to work! https://sourmash.readthedocs.io/en/add-nbsphinx/test-notebook.html

**Comment by ctb** [comment]
This is ready for initial commentary @luizirber @taylorreiter - thoughts very welcome!

**Comment by ctb** [comment]
Ready for review @luizirber @taylorreiter @bluegenes - once this is reviewed and OKed, I will update all the references to the `add/nbsphinx` branch to point to master, and then merge.

**Comment by taylorreiter** [comment]
+ I don't see the "Notebooks" section at sourmash.readthedocs.io
+ I like that you link to other notebooks at the bottom of each notebook
+ "sourmash: working with private collections of signatures" `In[2]` currently has output that indicates the signatures are already present, so the calculations do not occur. I can't find the `big_genomes` directory to see whether this is expected behavior (did `big_genomes` get switched to `genomes`?). If it is, maybe we should make a note of it in the notebook? 

Otherwise, LGTM!

**Comment by ctb** [comment]
Thanks @taylorreiter!

Curious - why are you looking for the Notebooks section? :) (I had that in earlier, but then I refactored.) Do I have a reference to it somewhere?

Thanks for the catch re big_genomes! It's created in the top of the notebook (`mkdir -p ...`) so that's no problem; I've redone the signature calculation and pushed updates.

If it looks good, I think it's ready to be approved (and then I'll go update all of the URLs before merging)

File: binder/environment.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+channels:
+  - bioconda
+  - defaults
+  - conda-forge
```
Comment (luizirber) [inline binder/environment.yml]: correct order: 
```
channels:
  - conda-forge
  - bioconda
  - defaults
```

Comment (ctb) [reply binder/environment.yml]: fixed in 674e5e8!

**Comment by ctb** [comment]
ok. maybe I didn't do this right? but it should be ready.

**Comment by ctb** [comment]
(that is, ready to be approved. I'll need to do the branch name editing before merging.)

**Comment by luizirber** [comment]
@ctb it's approved now. MacOS wheel building seems to be failing on Travis, probably related to https://github.com/joerick/cibuildwheel/issues/122 (will check)

**Comment by ctb** [comment]
thx all!

**Pull request merged**